File: apps/api/app/schemas/validation.py
from datetime import datetime

from pydantic import BaseModel


class ValidationRunCreate(BaseModel):
    command: str
    patch_artifact_id: int


class ValidationRunResponse(BaseModel):
    id: int
    session_id: int
    patch_artifact_id: int
    command: str
    exit_code: int
    stdout: str
    stderr: str
    analysis: str
    duration_ms: int
    created_at: datetime

    model_config = {"from_attributes": True}

File: packages/shared/src/types/validation.ts
export interface ValidationRun {
  id: number;
  sessionId: number;
  patchArtifactId: number;
  command: string;
  exitCode: number;
  stdout: string;
  stderr: string;
  analysis: string;
  durationMs: number;
  createdAt: string;
}

export interface RunValidationRequest {
  command: string;
  patchArtifactId: number;
}
